File: 01-data-preprocessing/correlation.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv("movies.csv")
df = df.dropna()
df["VOTES"] = df["VOTES"].str.replace(",", "").astype(int)

correlation = df["RATING"].corr(df["VOTES"])
print(f"Pearson Correlation: {correlation:.2f}")

sns.regplot(x="RATING", y="VOTES", data=df, scatter_kws={"alpha": 0.5})
plt.title("Scatter plot of Rating vs Votes")
plt.xlabel("Rating")
plt.ylabel("Votes")
plt.text(df["RATING"].min(), df["VOTES"].max(), f"Pearson Correlation: {correlation:.2f}", horizontalalignment="left")
plt.show()

File: 01-data-preprocessing/correlation.r
df <- read.csv("movies.csv")
df <- df[complete.cases(df$RATING, df$VOTES), ]
df$VOTES <- as.integer(gsub(",", "", df$VOTES))

correlation <- cor(df$RATING, df$VOTES, method = c("pearson"))
cat("Correlation coefficient:", round(correlation, 2), "\n")

plot(df$RATING, df$VOTES, main = "Scatter plot of Rating vs Votes", xlab = "Rating", ylab = "Votes")

fit <- lm(VOTES ~ RATING, data = df)
abline(fit, col = "blue")

text(min(df$RATING), max(df$VOTES), labels = paste("Pearson Correlation:", round(correlation, 2)), pos = 4)
